app/src/main/java/com/example/ecodigify/notifications/NotificationScheduler.kt

```diff
@@ -0,0 +1,89 @@
+package com.example.ecodigify.notifications
+
+import android.annotation.SuppressLint
+import android.app.AlarmManager
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import com.example.ecodigify.dataclass.Ingredient
+import java.util.TimeZone
+
+class NotificationScheduler(private val context: Context) {
+    init {
+        createNotificationChannel()
+    }
+
+    // Schedule a notification for a specific LocalDate (on that date at midnight)
+    fun scheduleNotificationFor(ingredient: Ingredient) {
+        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+
+        val notificationTimeMillis = ingredient.expirationDate
+            .atStartOfDay()
+            .atZone(TimeZone.getDefault().toZoneId())
+            .toInstant()
+            .toEpochMilli()
```

Comment:
```suggestion
        val notificationTimeMillis = ingredient.expirationDate
            .minusDays(2)
            .atStartOfDay()
            .atZone(TimeZone.getDefault().toZoneId())
            .toInstant()
            .toEpochMilli()
```
Sarebbero da mandare 2 giorni prima, bisogna vedere se mandarle anche gli altri giorni

Reply:
Questo per ora lo terrei così. Ho aggiunto una `TODO` per ricordarmi di farlo più dinamico. 

L'ideale sarrebbe una cosa tipo:
1. La setto una settimana prima 
2. Quando la notifica è stata vista, ce lo si segna da qualche parte e si fa uno scheduling di una notifica per tipo 2 giorni prima
3. Quando il coso è scaduto si manda un altra notifica

Reply:
Stando ai documenti non avevamo previsto una singola notifica 2gg prima?